src/ukify/ukify.py

```diff
@@ -519,60 +511,34 @@ def call_systemd_measure(uki, opts):
 
     # PCR measurement
 
-    to_measure = []
-    tflist = []
-
-    # First, pick up the sections we shall measure now */
-    for s in uki.sections:
-        if not s.measure:
-            continue
-
-        if s.content is not None:
-            to_measure.append(f"--{s.name.removeprefix('.')}={s.content}")
-        else:
-            raise ValueError(f"Don't know how to measure section {s.name}");
-
-    # And now iterate through the base profile and measure what we haven't measured above
-    if opts.measure_base is not None:
-        pe = pefile.PE(opts.measure_base, fast_load=True)
-
-        # Find matching PE section in base image
-        for base_section in pe.sections:
-            name = pe_strip_section_name(base_section.Name)
+    # First, pick up either the base sections or the profile specific sections we shall measure now
+    to_measure = {s.name: s for s in uki.sections[profile_start:] if s.measure}
 
+    # Then, if we're measuring a profile, lookup the missing sections from the base image.
+    if profile_start != 0:
+        for section in uki.sections:
             # If we reach the first .profile section the base is over
-            if name == ".profile":
+            if section.name == ".profile":
                 break
 
             # Only some sections are measured
-            if name not in measure_sections:
+            if not section.measure:
                 continue
 
             # Check if this is a section we already covered above
-            already_covered = False
-            for s in uki.sections:
-                if s.measure and name == s.name:
-                    already_covered = True
-                    break;
-
-            if already_covered:
+            if section.name in to_measure:
                 continue
 
-            # Split out section and use as base
-            tf = tempfile.NamedTemporaryFile()
-            tf.write(base_section.get_data(length=base_section.Misc_VirtualSize))
-            tf.flush()
-            tflist.append(tf)
-
-            to_measure.append(f"--{name.removeprefix('.')}={tf.name}")
+            to_measure[section.name] = section
 
     if opts.measure:
         pp_groups = opts.phase_path_groups or []
 
         cmd = [
             measure_tool,
             'calculate',
-            *to_measure,
+            *(f"--{s.name.removeprefix('.')}={s.content}"
+              for _, s in to_measure.items()),
```

Comment:
Taking the key out and not using it seems a bit unnecessary.
```suggestion
              for s in to_measure.values()),
```
I checked in the python 3.9 docs that insertion order is preserved for the dictionary views.